magicctapipe/scripts/lst1_magic/README.md

```diff
@@ -0,0 +1,226 @@
+- `merge_hdf_files.py` on MAGIC data to merge subruns and/or runs together.
+- `lst1_magic_event_coincidence.py` to find coincident events between MAGIC and LST-1, starting from DL1 data.
+- `lst1_magic_stereo_reco.py` to add stereo parameters to the DL1 data.
+- `lst1_magic_train_rfs.py` to train the RFs (energy, direction, classification) on train gamma MCs and protons.
+- `lst1_magic_dl1_stereo_to_dl2.py` to apply the RFs to stereo DL1 data (real and test MCs) and produce DL2 data.
+- `lst1_magic_create_irf.py` to create the IRF.
+- `lst1_magic_dl2_to_dl3.py` to create DL3 files, and `create_dl3_index_files.py` to create DL3 HDU and index files.
+
+From DL3 on, the analysis is done with gammapy.
+
+## Installation
+
+1) The very first step to reduce MAGIC-LST data is to have remote access/credentials to the IT Container, so provide one. Once you have it, the connection steps are the following:
+
+Authorized institute server (Client) &rarr;  ssh connection to CTALaPalma &rarr; ssh connection to cp01/02.
+
+2) Once connected to the IT Container, install magic-cta-pipe (e.g. in your home directory in the IT Container) with the following commands (if you have mamba installed, we recommend you to use it instead of conda. The installation process will be much faster.):
+
+```
+git clone -b Torino_auto_MCP https://github.com/cta-observatory/magic-cta-pipe.git
+cd magic-cta-pipe
+mamba env create -n magic-lst -f environment.yml
+mamba activate magic-lst
+pip install .
+```
+
+## Analysis
+
+During the analysis, some files are automatically produced by the scripts and are saved in your working directory. These files are necessary for the subsequent steps in the analysis chain. It is therefore mandatory to always launch the scripts from the same working directory so that the output files stored there can be correctly assigned as input files at the subsequent analysis steps.
+
+### DL0 to DL1
+
+In this step, we will convert the MAGIC and Monte Carlo (MC) Data Level (DL) 0 to DL1 (our goal is to reach DL3).
+
+In your working IT Container directory (e.g. /fefs/aswg/workspace/yourname/yourprojectname), open the magic-lst environment with the command `conda activate magic-lst` and update the file `config_general.yaml` according to your analysis.
+
+The file `config_general.yaml` must contain the telescope IDs, the directories with the MC data (mandatory only if NSB_matching = False), the data selection, and some information on the night sky background (NSB) level and software versions:
+
+```
+mc_tel_ids:
+    LST-1: 1
+    LST-2: 0
+    LST-3: 0
+    LST-4: 0
+    MAGIC-I: 2
+    MAGIC-II: 3
+
+directories:
+    workspace_dir : "/fefs/aswg/workspace/yourname/yourprojectname/"  # Output directory where all the data products will be saved.
+    # MC paths below are ignored if you set NSB_matching = true.
+    MC_gammas     : "/fefs/aswg/data/mc/DL0/LSTProd2/TestDataset/sim_telarray"  # set to "" if you don't want to process these Monte Carlo simulations.
```

Comment:
even we use preprocessed MCs, we will need at some point the RF files for DL2 data files and the IRF files for producing DL3 files. 
If those are going to be used from the general MC table, we will need to specify some MC-related settings: analysis period and processing version, unless the scripts will figure them out themselfs (important if the date range requested involves multiple periods)

Reply:
As for the analysis period, if you mean the MAGIC obs period (e.g., ST0317) the scripts will (?) estimate the right period for each observation (based on dates) and search for the MC of the right period for each run, so that you can process in the same analysis data collected over >2 years (indeed, in the previous version we created one folder per period since MAGIC DL1 and we matched data-MC periods based on the directories names). As for MC processing version, now there is only one version and it is hard-coded in the paths  in the DL1 to DL2 and DL2 to DL3 automatic scripts (still private scripts, but they exist ;) ), but we can for sure add an entry in the config file for this version

Reply:
ok, since this is for DL2 ==> DL3, and those scripts are not a part of this PR, I'm closing this for the moment (and will check it again once we reach to DL3)